### PR TITLE
docs(NODE-4794): Clean up migration guide for BSON v5

### PR DESCRIPTION
**Pull request by alexbevi**
### Double check the following

- [ ] Ran `npm run lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: docs/upgrade-to-v5.md [hunk 1]
```diff
@@ -26,15 +25,15 @@ for Node.js and web platforms.
...
+> **TL;DR**: Impacted APIs now return `Uint8Array` in web environments; Node.js environments are unaffected
 
-For those that use the BSON library on Node.js, there is no change - the BSON APIs will still return and accept instances of Node.js Buffer. Since we no longer depend on the Buffer web shim for compatibility with browsers, in non-Node.js environments a Uint8Array will be returned instead.
+For those that use the BSON library on Node.js, there is no change - the BSON APIs will still return and accept instances of Node.js `Buffer`. Since we no longer depend on the `Buffer` web shim for compatibility with browsers, in non-Node.js environments a `Uint8Array` will be returned instead.
 
-This allows the BSON library to be better at platform independence while keeping its behavior consistent cross platform. The Buffer shim served the library well but brought in more than was necessary for the concerns of the code here.
+This allows the BSON library to be more platform independent while keeping its behavior consistent cross platform. The `Buffer` shim served the library well but brought in more than was necessary for the concerns of the code here.
```
Comment (nbbeeken) [inline docs/upgrade-to-v5.md:32]:
```suggestion
This allows the BSON library to be more platform independent while keeping its behavior consistent cross platform. 
```

I wrote this but I think this sentence doesn't add valuable information

File: docs/upgrade-to-v5.md [hunk 2]
```diff
@@ -183,8 +184,8 @@ EJSON.stringify({}, { strict: false }); /* migrate to */ EJSON.stringify({}, { r
 
 ### The BSON default export has been removed.
 
-* If you import BSON commonjs style `const BSON = require('bson')` then the `BSON.default` property is no longer present.
-* If you import BSON esmodule style `import BSON from 'bson'` then this code will crash upon loading. **TODO: This is not the case right now but it will be after NODE-4713.**
+* If you import BSON `commonjs` style `const BSON = require('bson')` then the `BSON.default` property is no longer present.
+* If you import BSON `esmodule` style `import BSON from 'bson'` then this code will crash upon loading. **TODO: This is not the case right now but it will be after [NODE-4713](https://jira.mongodb.org/browse/NODE-4713).**
```
Comment (nbbeeken) [inline docs/upgrade-to-v5.md:188]:
```suggestion
* If you import BSON `esmodule` style `import BSON from 'bson'` then this code will crash upon loading.
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: docs/upgrade-to-v5.md (hunk 1 above)
Comment (alexbevi) [reply docs/upgrade-to-v5.md:32]: sgtm

**Comment by alexbevi** [comment]
Overall the docs was in really good shape. I just cleaned up a little formatting.

**Review by nbbeeken — APPROVED** [review]
Thanks for the cleanup! lgtm

File: docs/upgrade-to-v5.md [hunk 3]
```diff
@@ -26,15 +25,15 @@ for Node.js and web platforms.
...
+For those that use the BSON library on Node.js, there is no change - the BSON APIs will still return and accept instances of Node.js `Buffer`. Since we no longer depend on the `Buffer` web shim for compatibility with browsers, in non-Node.js environments a `Uint8Array` will be returned instead.
 
-This allows the BSON library to be better at platform independence while keeping its behavior consistent cross platform. The Buffer shim served the library well but brought in more than was necessary for the concerns of the code here.
+This allows the BSON library to be more platform independent while keeping its behavior consistent cross platform. 
 
 #### APIs impacted
 
-The following APIs now return Uint8Arrays when the library is loaded in an environment that does not define a global Node.js Buffer.
+The following APIs now return `Uint8Arrays` when the library is loaded in an environment that does not define a global Node.js Buffer.
```
Comment (addaleax) [inline docs/upgrade-to-v5.md:36]: for consistency?

```suggestion
The following APIs now return `Uint8Arrays` when the library is loaded in an environment that does not define a global Node.js `Buffer`.
```

File: docs/upgrade-to-v5.md [hunk 4]
```diff
@@ -242,9 +244,9 @@ BSON.deserialize(result)
 ### Exports and available bundles
 
 Most users should be unaffected by these changes, Node.js `require()` / Node.js `import` will fetch the corresponding BSON library as expected.
-And for folks using bundlers like, webpack or rollup a tree shakable es module bundle will be pulled in because of the settings in our package.json.
+And for folks using bundlers like webpack or rollup, a tree shakable ES module bundle will be pulled in because of the settings in our `package.json`.
 
-Our package.json defines the following `"exports"` settings.
+Our `package.json` defines the following `"exports"` settings.
 ```json
 {
```
Comment (nbbeeken) [inline docs/upgrade-to-v5.md:251]: This just needs an update to be accurate, it changed slightly after that PR:

```json
"main": "./lib/bson.cjs",
  "module": "./lib/bson.mjs",
  "exports": {
    "import": "./lib/bson.mjs",
    "require": "./lib/bson.cjs",
    "react-native": "./lib/bson.cjs",
    "browser": "./lib/bson.mjs"
  },
  ```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: docs/upgrade-to-v5.md (hunk 4 above)
Comment (alexbevi) [reply docs/upgrade-to-v5.md:251]: Updated

**Review by nbbeeken — APPROVED** [review]
Ty! lgtm!

**Pull request merged**